pandas/core/generic.py

```diff
@@ -61,6 +61,10 @@
         by : str or list of str
             Name or list of names to sort by""")
 
+# sentinel value to use as kwarg in place of None when None has special meaning
+# and needs to be distinguished from a user explicitly passing None.
```

Comment:
I would just use object() its more pythonic, we should remove the whole sentinel_factory() bizness.

Reply:
made the `sentinel_factory()` --> `object()` change here; planning to do the global removal of `sentinel_factory()` as a follow up so it can go into 0.25.0 and keep the 0.24.x changes as minimal as possible.

Reply:
xref #25074